File: src/main/java/dev/budget/reconciler/model/TransactionDifference.java
package dev.budget.reconciler.model;

public class TransactionDifference {
	private String mintDate;
	private String mintTransaction;
	private int mintCents;
	private String ynabDate;
	private String ynabTransaction;
	private int ynabCents;

	public String getMintDate() {
		return mintDate;
	}

	public void setMintDate(String mintDate) {
		this.mintDate = mintDate;
	}

	public String getMintTransaction() {
		return mintTransaction;
	}

	public void setMintTransaction(String mintTransaction) {
		this.mintTransaction = mintTransaction;
	}

	public int getMintCents() {
		return mintCents;
	}

	public void setMintCents(int mintCents) {
		this.mintCents = mintCents;
	}

	public String getYnabDate() {
		return ynabDate;
	}

	public void setYnabDate(String ynabDate) {
		this.ynabDate = ynabDate;
	}

	public String getYnabTransaction() {
		return ynabTransaction;
	}

	public void setYnabTransaction(String ynabTransaction) {
		this.ynabTransaction = ynabTransaction;
	}

	public int getYnabCents() {
		return ynabCents;
	}

	public void setYnabCents(int ynabCents) {
		this.ynabCents = ynabCents;
	}
}

File: src/main/scala/dev/budget/reconciler/model/TransactionDifference.scala
package dev.budget.reconciler.model

import com.fasterxml.jackson.core.`type`.TypeReference
import com.fasterxml.jackson.module.scala.JsonScalaEnumeration
import dev.budget.reconciler.model.TransactionDifferenceType.TransactionDifferenceType
import org.joda.time.LocalDate

class TransactionDifferenceTypeHolder extends TypeReference[TransactionDifferenceType.type]

case class TransactionDifference(
  mintId: String,
  mintDate: LocalDate,
  mintTransaction: String,
  mintCents: Long,
  ynabId: String,
  ynabDate: LocalDate,
  ynabTransaction: String,
  ynabCents: Long,
  @JsonScalaEnumeration(classOf[TransactionDifferenceTypeHolder])
  differenceType: TransactionDifferenceType)
